files/files.go

```diff
@@ -0,0 +1,153 @@
+package files
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+func ExtractAll(files []File) error {
```

Comment:
I suppose the intent is to fail fast and hard if we hit a single file that we issues with. Seems fine.